Mage/src/main/java/mage/abilities/dynamicvalue/common/CountersSourcePlayerCount.java

```diff
@@ -0,0 +1,65 @@
+package mage.abilities.dynamicvalue.common;
+
+import mage.abilities.Ability;
+import mage.abilities.dynamicvalue.DynamicValue;
+import mage.abilities.effects.Effect;
+import mage.counters.Counter;
+import mage.counters.CounterType;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+
+import java.util.UUID;
+
+public class CountersSourcePlayerCount implements DynamicValue {
+
+    private final CounterType counterType;
+
+    /**
+     * Number of counters of any type on the source permanent
+     */
+    public CountersSourcePlayerCount() {
+        this((CounterType) null);
+    }
+
+    /**
+     * Number of counters of the specified type on the source permanent
+     */
+    public CountersSourcePlayerCount(CounterType counterType) {
+        this.counterType = counterType;
+    }
+
+    protected CountersSourcePlayerCount(final CountersSourcePlayerCount countersCount) {
+        this.counterType = countersCount.counterType;
+    }
+
+    @Override
+    public int calculate(Game game, Ability sourceAbility, Effect effect) {
+        UUID controllerId = sourceAbility.getControllerId();
+        if (controllerId == null) {
```

Comment:
you don't need to null check this since you aren't calling a method on it - just inline it to the getPlayer call which is null checked